### ISSUE TITLE
Arrow Functions and compact method  cause false positive for UndefinedVariable bug

### ISSUE THREAD

**Issue by yokozawa0701** [sirbrillig/phpcs-variable-analysis#330]
Here's a simple example that generates UndefinedVariable for using the Arrow Functions and compact method.

```php
$something = fn ($arg) => compact('arg');  // Unused function parameter $arg. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable) 
```

I am using v2.11.19.
Thank you.

**Comment by sirbrillig** [comment]
Good find. I think this is happening because `processCompact()` calls `findVariableScope()` which has a special case for arrow function variables but needs the name of the variable itself, and we don't pass that (there could be many). Working on how to handle that case.

**Comment by sirbrillig** [comment]
This should be fixed by https://github.com/sirbrillig/phpcs-variable-analysis/pull/339
